opensrp-chw-referral/src/main/java/org/smartregister/chw/referral/activity/BaseIssueReferralActivity.java

```diff
@@ -0,0 +1,363 @@
+
+import com.google.gson.Gson;
+import com.rengwuxian.materialedittext.MaterialEditText;
+import com.wdullaer.materialdatetimepicker.date.DatePickerDialog;
+
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+import org.json.JSONObject;
+import org.smartregister.AllConstants;
+import org.smartregister.chw.referral.R;
+import org.smartregister.chw.referral.ReferralLibrary;
+import org.smartregister.chw.referral.contract.BaseIssueReferralContract;
+import org.smartregister.chw.referral.domain.MemberObject;
+import org.smartregister.chw.referral.domain.ReferralServiceIndicatorObject;
+import org.smartregister.chw.referral.interactor.BaseIssueReferralInteractor;
+import org.smartregister.chw.referral.model.AbstractIssueReferralModel;
+import org.smartregister.chw.referral.model.BaseIssueReferralModel;
+import org.smartregister.chw.referral.presenter.BaseIssueReferralPresenter;
+import org.smartregister.chw.referral.provider.ReferralServicesProvider;
+import org.smartregister.chw.referral.util.Constants;
+import org.smartregister.commonregistry.CommonPersonObject;
+import org.smartregister.commonregistry.CommonPersonObjectClient;
+import org.smartregister.commonregistry.CommonRepository;
+import org.smartregister.domain.Location;
+import org.smartregister.chw.referral.databinding.ActivityReferralRegistrationBinding;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import fr.ganfra.materialspinner.MaterialSpinner;
+import timber.log.Timber;
+
+/**
+ * Created by cozej4 on 2019-10-07.
+ *
+ * @cozej4 https://github.com/cozej4
+ */
+
+public class BaseIssueReferralActivity extends AppCompatActivity implements BaseIssueReferralContract.View {
+    protected BaseIssueReferralContract.Presenter presenter;
+    protected String BASE_ENTITY_ID;
+    protected List<String> SERVICE_IDS;
+    protected String ACTION;
+    protected String FORM_NAME;
+    protected TextView textViewName;
+    protected TextView textViewGender;
+    protected TextView textViewLocation;
+    protected TextView textViewUniqueID;
+    private AbstractIssueReferralModel viewModel;
+    private MaterialSpinner spinnerService;
+    private LinearLayout linearLayoutIndicators;
+    private View viewIndicator;
+    private MaterialEditText textViewAppointmentDate;
+    private AutoCompleteTextView autoCompleteTextViewFacilityName;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.BASE_ENTITY_ID = this.getIntent().getStringExtra("BASE_ENTITY_ID");
+        this.SERVICE_IDS = this.getIntent().getStringArrayListExtra(Constants.ACTIVITY_PAYLOAD.REFERRAL_SERVICE_IDS);
+        this.ACTION = this.getIntent().getStringExtra("ACTION");
+        this.FORM_NAME = this.getIntent().getStringExtra("REFERRAL_FORM_NAME");
```

Comment:
You can extract the strings for  BASE_ENTITY_ID, ACTION and FORM_NAME to constant like you have done for the SERVICE_IDS